src/main/kotlin/LocalBroker.kt

```diff
@@ -0,0 +1,39 @@
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+class LocalBroker: IMessageBroker {
+    var queues: MutableMap<QueueName, LocalQueue> = mutableMapOf()
+    override fun submitMessage(queueName: QueueName, message: Message) {
+        val q = queues.getOrPut(queueName) { LocalQueue() }
+        GlobalScope.launch {
+            delay(message.delayMs)
+            q.channel.trySend(message).getOrThrow()
```

Comment:
could you explain this line?

I might understand somthing incorrect but I see two problems in it:
1. trySend will always return Success result cause channel size is unlimited so getOrThrow will never throws
2. if getOrThrow will throw somehow the exception won't be catched by submitMessage so the only result of that throw will be an error in logs with hard readable stacktrace (cause of courutines scope)

I would suggest to replace `trySend(message).getOrThrow()` with just `send`
And if we want submitMessage to guarantee that the message is in the queue to make submitMessage `suspend`

Reply:
This is a copy-paste from here https://github.com/vchain-dev/signals-app/blob/ad25226d2178eb7d946b936bcf3956f3e7aaa3ed/signals-app/src/main/java/com/zamna/docprocessing/XUploadBuffer.kt#L62 actyally.